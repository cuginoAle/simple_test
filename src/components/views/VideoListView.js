import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import VideoCard from 'components/views/VideoCard';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  max-height: 100%;

  .viewportName {
    font-size: 3em;
    padding: 10px;
    background-color: #eee;
    flex-shrink: 0;
    box-shadow: 0 1px 4px rgba(0, 0, 0, 0.6);
    position: relative;
    z-index: 2;
    &:after {
      content: "Mobile";
      margin-left: 1em;
    }
  }

  .videoListUl {
    flex-grow: 1;
    display: flex;
    flex-wrap: wrap;
    scroll-snap-type: y mandatory;
    overflow-y: auto;
    max-height: 100%;

    > li {
      min-width: 100%;
      padding: 10px;
      scroll-snap-align: start;
    }
  }

  @media screen and (min-width: 600px) {
    .viewportName {
      &:after {
        content: "Tablet";
      }
    }
    .videoListUl > li {
      min-width: 50%;
    }
  }

  @media screen and (min-width: 1024px) {
    .viewportName {
      &:after {
        content: "Desktop";
      }
    }
    .videoListUl {
      overflow: auto;
      flex-wrap: nowrap;
      scroll-snap-type: x mandatory;
      > li {
        min-width: 500px;
      }
    }
  }
`;

const VideoListView = ({ className, videoList }) => {
  const classes = ['VideoListView'];
  if (className) classes.push(className);

  return (
    <Wrapper className={classes.join(' ')}>
      <h2 className="viewportName">Viewport size:</h2>
      <ul className="videoListUl">
        {videoList.map((videoDetails) => (
          <li key={videoDetails.key}>
            <VideoCard videoDetails={videoDetails} />
          </li>
        ))}
      </ul>
    </Wrapper>
  );
};

VideoListView.displayName = 'VideoListView';

VideoListView.propTypes = {
  className: PropTypes.string,
  videoList: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

VideoListView.defaultProps = {
  className: null,
};

export default VideoListView;
