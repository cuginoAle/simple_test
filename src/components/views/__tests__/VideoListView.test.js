import { render } from '@testing-library/react';
import React from 'react';
import VideoListView from '../VideoListView';

describe('VideoCard', () => {
  it('Should match spanshot for video', () => {
    const videoList = [{
      key: 'test',
      src: 'some url...',
    }];
    const app = render(<VideoListView videoList={videoList} />);

    expect(app).toMatchSnapshot();
  });
});
