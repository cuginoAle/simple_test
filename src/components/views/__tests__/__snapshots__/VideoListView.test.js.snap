// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard Should match spanshot for video 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKgILt frLybC VideoListView"
      >
        <h2
          class="viewportName"
        >
          Viewport size:
        </h2>
        <ul
          class="videoListUl"
        >
          <li>
            <div
              class="sc-dlfnbm rLhzC VideoCard"
            >
              <div
                class="sc-bdfBwQ cudsGs busySpinner"
              >
                <div
                  class="sc-gsTCUz bsFaMp"
                />
              </div>
              <div
                preload="auto"
                style="width: 100%; height: 100%;"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKgILt frLybC VideoListView"
    >
      <h2
        class="viewportName"
      >
        Viewport size:
      </h2>
      <ul
        class="videoListUl"
      >
        <li>
          <div
            class="sc-dlfnbm rLhzC VideoCard"
          >
            <div
              class="sc-bdfBwQ cudsGs busySpinner"
            >
              <div
                class="sc-gsTCUz bsFaMp"
              />
            </div>
            <div
              preload="auto"
              style="width: 100%; height: 100%;"
            />
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
